clang/lib/CodeGen/CGLoopInfo.h

```diff
@@ -60,11 +61,44 @@ struct LoopAttributes {
   /// Value for llvm.loop.interleave.count metadata.
   unsigned InterleaveCount;
 
-  /// Value for llvm.loop.ivdep.enable metadata.
-  bool SYCLIVDepEnable;
-
-  /// Value for llvm.loop.ivdep.safelen metadata.
-  unsigned SYCLIVDepSafelen;
+  // Struct to contain the IVDep info.
+  struct IVDepInfo {
+    unsigned SafeLen;
+    mutable llvm::SmallVector<std::pair<const ValueDecl *, llvm::MDNode *>, 4>
+        Arrays;
+    IVDepInfo(unsigned SL) : SafeLen(SL) {}
+    IVDepInfo(unsigned SL, const ValueDecl *A, llvm::MDNode *MD) : SafeLen(SL) {
+      Arrays.emplace_back(A, MD);
+    }
+
+    bool HasArray(const ValueDecl *Array) const {
+      return Arrays.end() != GetArrayPairItr(Array);
+    }
+
+    decltype(Arrays)::iterator GetArrayPairItr(const ValueDecl *Array) {
+      return find_if(Arrays,
+                     [Array](const auto &Pair) { return Pair.first == Array; });
+    }
+
+    decltype(Arrays)::iterator GetArrayPairItr(const ValueDecl *Array) const {
+      return find_if(Arrays,
+                     [Array](const auto &Pair) { return Pair.first == Array; });
+    }
+
+    void EraseArray(const ValueDecl *Array) {
```

Comment:
Please, align all of the functions names with the LLVM naming policy (camel case):
https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly

Reply:
Ah, thought the change had gone through already.  There was a long discussion on the mailing list to change functions to start with a capital letter, I thought it had been decided.